nautobot/extras/forms/mixins.py

```diff
@@ -394,6 +401,37 @@ def clean(self):
           verify that the requested "source" object does not have an existing RelationshipAssociation to
           a different destination object.
         """
+
+        required_relationships = Relationship.objects.get_required_for_model(self.instance)
+        for side, relations in required_relationships.items():
+            for relation in relations:
+
+                # Not enforcing required symmetric relationships
+                if relation.symmetric:
+                    continue
+
+                # Skip self-referencing
+                if ContentType.objects.get_for_model(self.instance) == getattr(
+                    relation, f"{relation.required_side}_type"
+                ):
+                    continue
+
+                model_name = getattr(relation, f"{side}_type").model_class()._meta.verbose_name
+                cr_field_name = f"cr_{relation.slug}__{relation.required_side}"
+
+                if not self.cleaned_data[cr_field_name]:
+                    if relation.type in [
+                        RelationshipTypeChoices.TYPE_ONE_TO_MANY,
+                        RelationshipTypeChoices.TYPE_MANY_TO_MANY,
+                    ]:
```

Comment:
Another case where I  think checking `relation.has_many()` would be more correct.